black.py

```diff
@@ -736,8 +736,11 @@ def format_str(src_contents: str, *, mode: FileMode) -> FileContent:
         for _ in range(after):
             dst_contents.append(str(empty_line))
         before, after = elt.maybe_empty_lines(current_line)
-        for _ in range(before):
-            dst_contents.append(str(empty_line))
+        # Black should not insert empty lines at the beginning
+        # of the file
+        if len(dst_contents) > 0:
```

Comment:
I think this logic would be better encapsulated inside `EmptyLineTracker`.

Reply:
Will refactor so I can move it!